docusaurus/docs/protocol/tokenomics/token_logic_modules.md

```diff
@@ -0,0 +1,301 @@
+
+**Prior to** processing each individual TLM, we need to understand if the amount claimed
+by the supplier adheres to the optimistic maxima set per the limits of the Relay Mining algorithm.
+
+Suppliers always have the option to over-service an application (**i.e. do free work**),
+in order to ensure high quality service in the network. This may lead to offchain
+reputation benefits (e.g. Gateways favoring them), but suppliers' on-chain rewards
+are always limited by the cumulative amounts applications' stakes (at session start; per service)
+and the number of suppliers in the session.
+
+:::note
+
+TODO_POST_MAINNET: After the Shannon upgrade, the team at Grove has a lot of ideas
+related to on-chain reputation, [supplier overlay networks](https://www.notion.so/buildwithgrove/Off-chain-Application-Stake-Tracking-6a8bebb107db4f7f9dc62cbe7ba555f7?pvs=4), and
+much more, all of which is out of scope for the initial implementation.
+
+:::
+
+```mermaid
+---
+title: "Token Logic Modules (pre) Processing"
+---
+flowchart TB
+    CSA(["Claim Settlement Amount (CSA)"])
+    MCA(["MaxClaimPerSupplier (MCA) <br> = (AppStake / NumSuppliersPerSession)"])
+    CC{"Is CSA > MCA?"}
+    Update(Broadcast Event <br> that SA = MCA)
+    SOAE{{Application Overserviced <br> Event}}
+    TLMP("Process TLMs (SA)")
+
+    CSA -- CSA --> CC
+    MCA -- MCA --> CC
+
+    Update -..-> SOAE
+    CC -- Yes --> Update
+    CC -- No<br>SA=CSA --> TLMP
+    Update -- SA=CSA --> TLMP
+
+    TLMP --SA--> TLMBEM[[TLM: Burn Equals Mint]]
+    TLMP --SA--> TLMGI[[TLM: Global Inflation]]
+    TLMP --SA--> TLMGIRR[[TLM: Global Inflation Reimbursement Request]]
+
+    classDef tlm fill:#54ebd5,stroke:#333,stroke-width:2px;
+    classDef question fill:#e3db6d,stroke:#333,stroke-width:2px;
+    classDef event fill:#e8b761,stroke:#333,stroke-width:2px;
+
+    class TLMBEM,TLMGI,TLMGIRR tlm;
+    class SOAE event;
+    class CC question;
+```
+
+## TLM: Mint=Burn (MEB)
+
+_tl;dr The transfer of tokens from the applications to the suppliers based on the amount of work received and provided respectively._
+
+The `Mint=Burn` TLM is, _theoretically_, the only TLM necessary once the network
+reaches maturity in the far future.
+
+The same number of tokens minted to the **supplier module** is burned from
+the **application module**. The stake (in escrow) owned by the application which is
+paying for work is reduced and the rewards are distributed to the supplier and its
+revenue shareholder addresses.
+
+```mermaid
+---
+title: "Token Logic Module: Mint=Burn"
+---
+flowchart TD
+    SA(["Settlement Amount (SA)"])
+
+    SA -- Mint SA coins --> SM
+    SA -- Burn SA coins--> AM
+
+    subgraph SO[Supplier Operations]
+        SM[[Supplier Module]]
+        SK[(Supplier Keeper)]
+        SD{Distribute SA coins}
+        RSA[Other Revenue Share Addresses]
+
+        SM -.- SK
+        SK -. Burn if proof <br> is missing or invalid .-> OA
+        SD -->|% Distribution <br> Increase Balance| OA
```

Comment:
I would remove `OA` and `OPA` from distribution increase balance.

While some configurations distribute to them, it's not always the case. Reward distribution always looks at the shareholders list.

Reply:
Please TAL at the updated diagrams. Made a significant number of changes.